lib/imgix/rails/picture_tag.rb

```diff
@@ -18,10 +18,9 @@ def render
         source_tag_opts = opts[:tag_options] || {}
         source_url_params = opts[:url_params] || {}
         widths = opts[:widths]
-        if opts.except(:tag_options, :url_params, :widths).length > 0
-          ActiveSupport::Deprecation.warn('use :tag_options, :url_params, :widths instead.')
-          source_tag_opts = opts.slice!(*self.class.available_parameters).except(:widths)
-          source_url_params = opts.slice(*self.class.available_parameters).except(:widths)
+        unsupported_opts = opts.except(:tag_options, :url_params, :widths)
```

Comment:
Can we pull this out into a separate PR?

Reply:
This is heavily tied to the legal parameters code. If keys other than these three are specified, they used to be separated into URL params/tag options based on the list of legal parameters. Because we don't have a list of legal parameters anymore, segregation isn't possible. So this code raises an error with a list of keys that aren't supported.

Reply:
Right, okay. I understand now.